pkg/controller/atlas/provider.go

```diff
@@ -108,17 +103,63 @@ func (p *ProductionProvider) Client(ctx context.Context, secretRef *client.Objec
 
 	c, err := mongodbatlas.New(httpClient, mongodbatlas.SetBaseURL(p.domain), mongodbatlas.SetUserAgent(operatorUserAgent()))
 
-	return c, secretData[orgIDKey], err
+	return c, secretData.OrgID, err
 }
 
-func validateSecretData(secretData map[string]string) ([]string, bool) {
-	var missingFields []string
-	requiredKeys := []string{orgIDKey, publicAPIKey, privateAPIKey}
+func (p *ProductionProvider) SdkClient(ctx context.Context, secretRef *client.ObjectKey, log *zap.SugaredLogger) (*admin.APIClient, string, error) {
+	secretData, err := getSecrets(ctx, p.k8sClient, secretRef, &p.globalSecretRef)
+	if err != nil {
+		return nil, "", err
+	}
 
-	for _, key := range requiredKeys {
-		if _, ok := secretData[key]; !ok {
-			missingFields = append(missingFields, key)
-		}
+	// TODO review we need add a custom logger to http client
+	//httpClientWithCustomLogger := http.DefaultClient
+	//err = httputil.LoggingTransport(log)(http.DefaultClient)
+	//if err != nil {
+	//	return nil, "", err
+	//}
```

Comment:
don't check in commented code.

Reply:
I left a TODO to review this in the upcoming PRs because the tests don't care about logging in the HTTP client, but there was a config for it in the operator. I'll check they are relevant to the operator and decide whether to keep or remove for real.

Reply:
If we are absolutely sure that this comment goes away in a short-term subsequent PR this sounds acceptable. However, if we do know that this is going to be addressed in a subsequent PR I would still advocate in not checking in commented code but rather leave it in a local branch and address it in a subsequent PR.

Reply:
I assure you this code will either be added/removed in subsequent PRs. This function needs to be revisited because it will be replacing the current one in every controller we have.